app/services/puppet_fact_parser.rb

```diff
@@ -138,6 +154,21 @@ def get_facts_for_interface(interface)
     iface_facts
   end
 
+  def get_facts_for_interface(interface)
+    return get_facts_for_interface_legacy(interface) if facterversion[0] < 3
+    interface_fact = facts.dig(:networking, :interfaces, interface) || {}
+    iface_facts = interface_fact.each_with_object([]) do |(name, value), facts|
+      facts << [interfaces_attribute_map(name), value] if interfaces_attribute_map(name)
+    end
+    iface_facts = HashWithIndifferentAccess[iface_facts]
+    logger.debug { "Interface #{interface} facts: #{iface_facts.inspect}" }
+    iface_facts
+  end
+
+  def facterversion
+    facts[:facterversion].split('.').map(&:to_i)
```

Comment:
agree, will make it return empty array when `facterversion`is not reported. In that case, code will use legacy facts (v2)  because they are available in all versions of facter..